### PR TITLE
fix : add config check for open and closed range settings

### PR DESCRIPTION
**Pull request by foxspy**
issue: https://github.com/milvus-io/milvus/issues/37269

### CODE REVIEW

File: include/knowhere/config.h [hunk 1]
```diff
@@ -115,6 +115,7 @@ struct Entry<CFG_FLOAT> {
     std::optional<CFG_FLOAT::value_type> default_val;
     uint32_t type;
     std::optional<std::pair<CFG_FLOAT::value_type, CFG_FLOAT::value_type>> range;
+    std::optional<std::pair<bool, bool>> range_included;
```
Comment (alexanderguzhva) [inline include/knowhere/config.h:118]: please consider packing range into a structure `Range`, which would contain all the logic, including something like `bool Range::within(float value) {}` and `std::string Range::to_string() {}` (which produces a string representation)

Comment (foxspy) [reply include/knowhere/config.h:118]: Good idea~ I will apply it

**Pull request merged**